File: Hydrogene/Toffee/OS X/NSMenu/NSMenu/MainWindowController.cs
namespace NSMenu
{
	using AppKit;

	[IBObject]
	public class MainWindowController : NSWindowController
	{

    [IBOutlet]
    public NSTextField messageLabel;

		public override id init()
		{
			this = base.initWithWindowNibName("MainWindowController");
			return this;
		}

    public void sayHello()
    {
        this.messageLabel.stringValue = "Hello";
    }


    public void sayGoodBye()
    {
        this.messageLabel.stringValue = "Goodbye";
    }

	

	}
}

File: Oxygene/Toffee/OS X/NSMenu/NSMenu/MainWindowController.pas
namespace NSMenu;

interface

uses
  AppKit,
  Foundation;

type
  [IBObject]
  MainWindowController = public class(NSWindowController)
  private
  protected
  public

    [IBOutlet] messageLabel : NSTextField;
    method init: id; override;
    method sayHello;
    method sayGoodBye;
  end;

implementation

method MainWindowController.init: id;
begin
  self := inherited initWithWindowNibName('MainWindowController');
  result := self;
end;


method MainWindowController.sayHello;
begin
   self.messageLabel.stringValue := "Hello"
end;

method MainWindowController.sayGoodBye;
begin
  self.messageLabel.stringValue := "GoodBye"
end;



end.
